tokio-fs/src/file/metadata.rs

```diff
@@ -0,0 +1,39 @@
+use super::File;
+
+use futures::{Future, Poll};
+
+use std::fs::File as StdFile;
+use std::fs::Metadata;
+use std::io;
+
+const POLL_AFTER_RESOLVE: &str = "Cannot poll MetadataFuture after it resolves";
+
+/// Future returned by `File::metadata` and resolves to a `(Metadata, File)` instance.
+#[derive(Debug)]
+pub struct MetadataFuture {
+    file: Option<File>,
```

Comment:
Why not make the `Option<&mut File>` and return `Metadata` as result of the `Future` implementation?

Reply:
Addressed in [previous comment](https://github.com/tokio-rs/tokio/pull/385#discussion_r194213846).